src/Microsoft.TemplateEngine.Cli/HelpAndUsage/HelpForTemplateResolution.cs

```diff
@@ -8,82 +8,69 @@
 using Microsoft.TemplateEngine.Abstractions;
 using Microsoft.TemplateEngine.Cli.CommandParsing;
 using Microsoft.TemplateEngine.Utils;
+using Microsoft.TemplateEngine.Cli.TemplateResolution;
+using System.Text;
 
 namespace Microsoft.TemplateEngine.Cli.HelpAndUsage
 {
     internal static class HelpForTemplateResolution
     {
-        public static CreationResultStatus CoordinateHelpAndUsageDisplay(TemplateListResolutionResult templateResolutionResult, IEngineEnvironmentSettings environmentSettings, INewCommandInput commandInput, IHostSpecificDataLoader hostDataLoader, ITelemetryLogger telemetryLogger, TemplateCreator templateCreator, string defaultLanguage, bool showUsageHelp = true)
+        public static CreationResultStatus CoordinateHelpAndUsageDisplay(ListOrHelpTemplateListResolutionResult templateResolutionResult, IEngineEnvironmentSettings environmentSettings, INewCommandInput commandInput, IHostSpecificDataLoader hostDataLoader, ITelemetryLogger telemetryLogger, TemplateCreator templateCreator, string defaultLanguage, bool showUsageHelp = true)
```

Comment:
I'm still considering to split them to separate classes under same base class and probably also inherit `TemplateListResolutionResult` from it.  However it needs extra time and not sure if it's right moment to do it.
As a simple fix I may suggest `DisplayTemplateListResolutionResult` (display should fit both help and list) or just `HelpTemplateListResolutionResult` as list is considered help in many places.

Reply:
`ListOrHelpTemplateListResolutionResult` has two `List` inside. Is this necessary or can we omit one of those?

Reply:
this actually refers to my other comment below - implementation of the classes that used `TemplateListResolutionResult` and uses new `ListOrHelpTemplateListResolutionResult` class assumes having the list returned and not enumerable. 

Again it can be refactored probably along with counts cached not to enumerate each time just to get count, however it requires bigger effort.